custom-modifiers/waitforelementtobevisiblemodifier/src/main/java/WaitForElementToBeVisibleModifier.java

```diff
@@ -0,0 +1,77 @@
+import com.cognifide.aet.communication.api.metadata.CollectorStepResult;
+import com.cognifide.aet.job.api.collector.CollectorJob;
+import com.cognifide.aet.job.api.exceptions.ParametersException;
+import com.cognifide.aet.job.api.exceptions.ProcessingException;
+import org.openqa.selenium.By;
+import org.openqa.selenium.StaleElementReferenceException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.FluentWait;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.concurrent.TimeUnit;
+
+public class WaitForElementToBeVisibleModifier implements CollectorJob {
```

Comment:
You might not know about that, but those modifiers are perfect example of [`WebElementsLocatorParams`](https://github.com/Cognifide/aet/blob/master/core/jobs/src/main/java/com/cognifide/aet/job/common/modifiers/WebElementsLocatorParams.java). You may skip writing a lot of code here by extending it (see [`Click Modifier`](https://github.com/Cognifide/aet/blob/master/core/jobs/src/main/java/com/cognifide/aet/job/common/modifiers/click/ClickModifier.java) as example) - especially setting and validating params.